File: Python/Find All Numbers Disappeared in an Array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for num in nums:
            index = abs(num) - 1
            nums[index] *= (-1 if nums[index] > 0 else 1)
        return [i + 1 for i in xrange(len(nums)) if nums[i] > 0]

File: cpp/Find All Numbers Disappeared in an Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        for (int i = 0; i < nums.size(); ++i) {
            int idx = abs(nums[i]) - 1;
            if (nums[idx] > 0) {
                nums[idx] *= -1;
            }
        }
        vector<int> result;
        for (int j = 0; j < nums.size(); ++j) {
            if (nums[j] > 0) {
                result.push_back(j + 1);
            }
        }
        return result;

    }
};
